File: generated_tests/pascal/tests/terminal_tests.pas
uses SplashKit, TestFramework, ../Helpers;
type
    TTestTerminal = class(TTestCase)
    protected
        procedure Setup; override;
    end;
    procedure TestReadCharIntegration;
begin
    WriteLine();
    Write('Enter the letter A');
    result := ReadChar();
    AssertEquals('A', result);
end;
procedure TestReadLineIntegration;
begin
    WriteLine();
    Write('Enter the text: Test Input');
    result := ReadLine();
    AssertEquals('Test Input', result);
end;
procedure TestTerminalHasInputIntegration;
begin
    WriteLine();
    Write('Enter some text: Some Input');
    result := TerminalHasInput();
    AssertTrue(result);
end;
procedure TestWriteCharIntegration;
begin
    WriteLine();
    Write('Should print A:');
    Write('A');
    WriteLine();
end;
procedure TestWriteDoubleIntegration;
begin
    WriteLine();
    Write('Should print 3.14:');
    Write(3.14);
    WriteLine();
end;
procedure TestWriteIntIntegration;
begin
    WriteLine();
    Write('Should print 42:');
    Write(42);
    WriteLine();
end;
procedure TestWriteIntegration;
begin
    WriteLine();
    Write('Should print Test String:');
    Write('Test String');
    WriteLine();
end;
procedure TestWriteLineCharIntegration;
begin
    WriteLine();
    Write('Should print A:');
    WriteLine('A');
end;
procedure TestWriteLineEmptyIntegration;
begin
    WriteLine();
    Write('Should print empty line:');
    WriteLine();
end;
procedure TestWriteLineDoubleIntegration;
begin
    WriteLine();
    Write('Should print 3.14:');
    WriteLine(3.14);
end;
procedure TestWriteLineIntIntegration;
begin
    WriteLine();
    Write('Should print 42:');
    WriteLine(42);
end;
procedure TestWriteLineIntegration;
begin
    WriteLine();
    Write('Should print Test Line:');
    WriteLine('Test Line');
end;
end;

procedure RegisterTests;
begin
    #<Proc:0x00007faa116e2450 /mnt/c/Users/Noahc/Documents/aYear_2_semester_2/TeamProject/GitHubRepo/splashkit_test_generator/test_generator/config/languages/pascal_config.rb:138 (lambda)>
end;

File: generated_tests/csharp/tests/terminal_tests.cs
using Xunit;
using SplashKitSDK;
using static SplashKitSDK.SplashKit;
using HttpMethod = SplashKitSDK.HttpMethod;
namespace SplashKitTests
{
    public class TestTerminal
    {
        [Fact]
        public void TestReadCharIntegration() {
            WriteLine();
            Console.WriteLine("Enter the letter A");
            var result = ReadChar();
            Assert.Equal('A', result);
        }
        [Fact]
        public void TestReadLineIntegration() {
            WriteLine();
            Console.WriteLine("Enter the text: Test Input");
            var result = ReadLine();
            Assert.Equal("Test Input", result);
        }
        [Fact]
        public void TestTerminalHasInputIntegration() {
            WriteLine();
            Console.WriteLine("Enter some text: Some Input");
            var result = TerminalHasInput();
            Assert.True(result);
        }
        [Fact]
        public void TestWriteCharIntegration() {
            WriteLine();
            Console.WriteLine("Should print A:");
            Write('A');
            WriteLine();
        }
        [Fact]
        public void TestWriteDoubleIntegration() {
            WriteLine();
            Console.WriteLine("Should print 3.14:");
            Write(3.14);
            WriteLine();
        }
        [Fact]
        public void TestWriteIntIntegration() {
            WriteLine();
            Console.WriteLine("Should print 42:");
            Write(42);
            WriteLine();
        }
        [Fact]
        public void TestWriteIntegration() {
            WriteLine();
            Console.WriteLine("Should print Test String:");
            Write("Test String");
            WriteLine();
        }
        [Fact]
        public void TestWriteLineCharIntegration() {
            WriteLine();
            Console.WriteLine("Should print A:");
            WriteLine('A');
        }
        [Fact]
        public void TestWriteLineEmptyIntegration() {
            WriteLine();
            Console.WriteLine("Should print empty line:");
            WriteLine();
        }
        [Fact]
        public void TestWriteLineDoubleIntegration() {
            WriteLine();
            Console.WriteLine("Should print 3.14:");
            WriteLine(3.14);
        }
        [Fact]
        public void TestWriteLineIntIntegration() {
            WriteLine();
            Console.WriteLine("Should print 42:");
            WriteLine(42);
        }
        [Fact]
        public void TestWriteLineIntegration() {
            WriteLine();
            Console.WriteLine("Should print Test Line:");
            WriteLine("Test Line");
        }
    }
}
